packages/flutter_tools/lib/src/commands/devices.dart

```diff
@@ -34,6 +34,13 @@ class DevicesCommand extends FlutterCommand {
         'No devices detected.\n\n'
         'If you expected your device to be detected, please run "flutter doctor" to diagnose\n'
```

Comment:
If we're giving them diagnostic messages inline, does it still make sense to tell them to run `flutter doctor`?  Maybe move this message into an `else` block and only show it if there are no inline diagnostic messages...

Reply:
It's a bit of duplication, but we don't know exactly what the diagnostic information will show - it'll be coming from the native toolchain. I'd like to have slightly too much message here, then possibly just a short or cryptic line from the native tool. It may be possible to make this message more terse though?